addons/sale_stock/models/sale_order.py

```diff
@@ -424,10 +424,10 @@ def _prepare_procurement_values(self, group_id=False):
             'partner_id': self.order_id.partner_shipping_id.id,
             'company_id': self.order_id.company_id,
         })
-        for line in self.filtered("order_id.commitment_date"):
-            date_planned = fields.Datetime.from_string(line.order_id.commitment_date) - timedelta(days=line.order_id.company_id.security_lead)
+        if self.order_id.commitment_date:
+            date_planned = fields.Datetime.from_string(self.order_id.commitment_date) - timedelta(days=self.order_id.company_id.security_lead)
```

Comment:
you can also remove the fields.Datetime.from_string, `self.order_id.commitment_date` is already a datetime